src/transforms.ts

```diff
@@ -141,6 +150,119 @@ function kebabToSnake<T>(obj: T): DeepKebabToSnake<T> {
+  for (const [key, value] of searchParams) {
+    transformed.append(transformKey(key), value)
+  }
+  return transformed
+}
+
+function transformQuery(
+  query: SearchParams,
+  transformKey: (str: string) => string,
+) {
+  if (Array.isArray(query)) {
+    return query.map(([key, value]) => [transformKey(key), value])
+  }
+
+  if (query instanceof URLSearchParams) {
+    return transformSearchParams(query, transformKey)
+  }
+
+  if (typeof query === 'string') {
+    const searchParams = new URLSearchParams(query)
+    return transformSearchParams(searchParams, transformKey).toString()
+  }
+
+  if (typeof query === 'object') return deepTransformKeys(query, transformKey)
+
+  return query
+}
+
+function transformBody(
+  body: EnhancedBodyInit | undefined,
+  transformKey: (str: string) => string,
+) {
+  if (body instanceof URLSearchParams) {
+    return transformSearchParams(body, transformKey)
+  }
+
+  if (body instanceof FormData) {
+    return transformFormData(body, transformKey)
+  }
+
+  return deepTransformKeys(body, transformKey)
+}
+
+/**
+ * Creates a requestTransformer to use with makeService or makeFetcher that
+ * will deeply transform the keys of the query and the body of the request.
+ * @param transformKey a function that receives a key and transforms it
+ * @returns a RequestTransformer function
+ * @example const requestTransformer = makeRequestTransformer((key) =>
+ *   key.toUpperCase(),
+ * )
+ * const service = makeService('https://api.com', { requestTransformer })
+ * // This will uppercase all keys of a request's query and body
+ */
+function makeRequestTransformer(
+  transformKey: (str: string) => string,
+): RequestTransformer {
+  return (request) => {
+    const query = transformQuery(request.query, transformKey)
+    const body = transformBody(request.body, transformKey)
+
+    return { ...request, query, body }
+  }
+}
+
+const kebabRequest = makeRequestTransformer(toKebabCase)
+const snakeRequest = makeRequestTransformer(toSnakeCase)
+const camelRequest = makeRequestTransformer(toCamelCase)
+
+/**
+ * Creates a responseTransformer to use with makeService or makeFetcher that
+ * will deeply transform the keys of the JSON body of the response.
+ * @param transformKey a function that receives a key and transforms it
+ * @returns a ResponseTransformer function
+ * @example const responseTransformer = makeResponseTransformer((key) =>
+ *   key.toUpperCase(),
+ * )
+ * const service = makeService('https://api.com', { responseTransformer })
+ * // This will uppercase all keys of the response's JSON body
+ */
+function makeResponseTransformer(
+  transformKey: (str: string) => string,
+): ResponseTransformer {
+  const getJson: GetJson =
+    (response) =>
+    async <T = unknown>(schema?: Schema<T>): Promise<T> => {
+      const json = deepTransformKeys(await response.json(), transformKey)
+      return schema ? schema.parse(json) : (json as T)
+    }
+
+  return (response) => typedResponse(response, { getJson })
```

Comment:
wouldn't it be simpler to apply the transformation to the body? (specially since we already need to wait in the json case).

In this way you would avoid any changes to `typedResponse` and also make it easier to troubleshoot since the (zod) parsing of the body would always happen in the same function.

Reply:
The [body is a ReadableStream](https://developer.mozilla.org/en-US/docs/Web/API/Response/body) that can stream many forms of content. By only applying the transformations to the `json` function, we allow for other kinds of bodies. What implementation did you have in mind?

Reply:
we could create a new Response instance. Copying the body and headers.

Reply:
since the transformation above sho-uld be applied only once we call `json` on the `TypedResponse` your current implementation makes more sense.